_guidelines-v4/09-textencoding/01-text/01-textdivisions.md

```diff
@@ -21,8 +20,10 @@ The following example shows the encoding of a preface translated into three diff
 
 {% include mei example="text/text-sample359.xml" valid="" %}
 
-Having said that {% include link elem="div" %} identifies any structural organization of text, it is often helpful to distinguish the typology of division. The attributes **@type** and **@subtype** can be used for this purpose. It is required that **@type** be present when **@subtype** is used, though their values can be freely set by the encoder.
+Having said that {% include link elem="div" %} identifies any structural organization of text, it is often helpful to distinguish the typology of division. The **@type** attribute can be used for this purpose.
+
+{% include desc atts="div/type" %}
```

Comment:
That's actually a very good catch. Here, `@type` isn't coming from a class, but is directly defined on `<div>`. This situation isn't handled by `_includes/desc` and `_includes/desc-item` yet. Not sure on how to proceed – of course we can work around the situation, but it would be better to fix those includes, right? Any suggestions, @ahankinson or @lpugin?

Reply:
we can change the include template, but what should it point to?

Reply:
Attributes defined directly within element do cause an overhead when building tools around the schema. Wouldn't it a better practice to make sure all attributes are defined only in attribute classes? Or is this unrealistic?

Reply:
It would be something to consider for future versions – can't get this out of v4 anymore. Problem is that here, an otherwise generic attribute (`@type` from `att.typed`) gets a more specific definition on `<div>`, with suggested values. Putting that into a separate class is certainly possible, but doesn't necessarily make things easier (could even break customizations that actively include attribute classes and still rely on `div/@type`). Again, I'm not opposed to changing things in the schema, but not for v4 anymore – those would break compatibility. Here, we need to either change the way we document things, or we need to adjust tools.

Reply:
> Here, we need to either change the way we document things, or we need to adjust tools.

I would vote for the first ;-). Unless changing the tools is easy. I guess the /div@type documentation is generated by the XSLT. What and where does it put it?

Reply:
No, actually this happens inside Jekyll. It already pulls the right text for display, but generates an incorrect link. The `{% include desc atts="div/type" %}` calls for an attribute class `att.div` with the attribute `type`, which obviously isn't there. We would need something like `{% include desc elem="div" atts="type" %}` or so, which generates the same visual HTML output, but with a different link (to the element, in this case)